### PR TITLE
Put the shadow of the header in the foreground.

### PR DESCRIPTION
**Pull request by MicheleK17**
Ticket: #336
Change-Id: I7d70d41fa3caf435520dbc5854234c9829c119e8

### CODE REVIEW

**Review by GerberI — CHANGES_REQUESTED** [review]
Commit: Put the shadow of the header in the foreground.
Reject: Please don't use global z-indices. Use stacking context instead: https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/The_stacking_context
So in your case you should add a z-index: 0 to parents and ensure they build an own stacking context.

File: src/app/Main/Header/Header.component.tsx [hunk 1]
```diff
@@ -16,6 +16,7 @@ const headerStyle = css`
     width: 100%;
     height: 100%;
     display: flex;
+    z-index: 0;
```
Comment (GerberI) [inline src/app/Main/Header/Header.component.tsx:19]: still global, please remove

File: src/app/Main/Header/Header.component.tsx [hunk 2]
```diff
@@ -28,7 +29,7 @@ const mainHeaderStyle = (open: boolean): SerializedStyles => css`
     justify-content: space-between;
     box-sizing: border-box;
     box-shadow: ${open ? "0 8px 6px -6px grey" : "none"};
-    z-index: 10;
+    z-index: 1;
```
Comment (GerberI) [inline src/app/Main/Header/Header.component.tsx:32]: still global please remove.
The header should not be responsible of it's z position. It is the responsibility of the parent component (some layout?) to arrange the components. See also comment in QueryPageLayout how to do that.

File: src/app/Main/Body/QueryPage/QueryPage-Layout.component.tsx [hunk 3]
```diff
@@ -17,10 +17,10 @@ const subHeaderStyle = css`
     background-color: ${bfrPrimaryPalette[300]};
     box-sizing: border-box;
     box-shadow: 0 8px 6px -6px grey;
-    z-index: 10;
+    z-index: 1;
```
Comment (GerberI) [inline src/app/Main/Body/QueryPage/QueryPage-Layout.component.tsx:20]: Nope
This component should go like:
mainStyle is the parent and should establish a new stacking context. As documented use:
mainStyle: position: relative (also the default but it must be set to establish the stacking), z-index: 0
subHeaderStyle and queryPageStyle are your components you want to arrange in the stacking-context, so:
subHeaderStyle: z-index: 1
queryPageStyle: z-index: 0

File: src/app/Main/Body/QueryPage/QueryPage-Layout.component.tsx [hunk 4]
```diff
@@ -17,10 +17,10 @@ const subHeaderStyle = css`
...
 `
-
 const queryPageStyle = css`
+    z-index: 0;
```
Comment (GerberI) [inline src/app/Main/Body/QueryPage/QueryPage-Layout.component.tsx:23]: Nope

**Review by GerberI — CHANGES_REQUESTED** [review]
Commit: Use stacking context
Not correct.
A while ago i thought about to make it mandatory for developers to add a position: relative/absolute and z-index: 0 to each components main html element. This would make it impossible to mess up the z-indices.

**Comment by MicheleK17** [comment]
Resolved on new branch "story/336_new"

**Pull request closed without merging**